ctru-rs/src/services/fs.rs

```diff
@@ -124,953 +111,6 @@ pub enum ArchiveID {
-/// # Examples
-///
-/// Opening a file to read:
-///
-/// ```
-/// # let _runner = test_runner::GdbRunner::default();
-/// use ctru::services::fs::{Fs, OpenOptions};
-///
-/// let mut fs = Fs::new().unwrap();
-/// let mut sdmc_archive = fs.sdmc().unwrap();
-/// let result = OpenOptions::new()
-///             .read(true)
-///             .archive(&sdmc_archive)
-///     .open("foo.txt");
-///
-/// assert!(result.is_err());
-/// ```
-///
-/// Opening a file for both reading and writing, as well as creating it if it
-/// doesn't exist:
-///
-/// ```
-/// # let _runner = test_runner::GdbRunner::default();
-/// use ctru::services::fs::{Fs, OpenOptions};
-///
-/// let mut fs = Fs::new().unwrap();
-/// let mut sdmc_archive = fs.sdmc().unwrap();
-/// let file = OpenOptions::new()
-///             .read(true)
-///             .write(true)
-///             .create(true)
-///             .archive(&sdmc_archive)
-///     .open("/foo.txt")
-///             .unwrap();
-/// ```
-#[derive(Clone, Copy, Debug, Default, PartialEq, Eq)]
-pub struct OpenOptions {
-    read: bool,
-    write: bool,
-    append: bool,
-    truncate: bool,
-    create: bool,
-    arch_handle: u64,
-}
-
-/// Iterator over the entries in a directory.
-///
-/// This iterator is returned from the [`read_dir`] function and
-/// will yield instances of [`Result<DirEntry, i32>`]. Through a [`DirEntry`]
-/// information like the entry's path and possibly other metadata can be
-/// learned.
-///
-/// # Errors
-///
-/// This Result will return Err if there's some sort of intermittent IO error
-/// during iteration.
-pub struct ReadDir<'a> {
-    handle: Dir,
-    root: Arc<PathBuf>,
-    arch: &'a Archive,
-}
-
-/// Entries returned by the [`ReadDir`] iterator.
-///
-/// An instance of `DirEntry` represents an entry inside of a directory on the
-/// filesystem. Each entry can be inspected via methods to learn about the full
-/// path or possibly other metadata.
-pub struct DirEntry<'a> {
-    entry: ctru_sys::FS_DirectoryEntry,
-    root: Arc<PathBuf>,
-    arch: &'a Archive,
-}
-
-#[doc(hidden)]
-struct Dir(u32);
-
-#[doc(hidden)]
-unsafe impl Send for Dir {}
-#[doc(hidden)]
-unsafe impl Sync for Dir {}
-
-impl Fs {
-    /// Initialize a new service handle.
-    ///
-    /// # Errors
-    ///
-    /// This function will return Err if there was an error initializing the
-    /// FS service, which in practice should never happen unless there is
-    /// an error in the execution environment (i.e. the homebrew launcher
-    /// somehow fails to provide fs:USER permissions)
-    ///
-    /// ctrulib services are reference counted, so this function may be called
-    /// as many times as desired and the service will not exit until all
-    /// instances of Fs drop out of scope.
-    pub fn new() -> crate::Result<Fs> {
-        unsafe {
-            let r = ctru_sys::fsInit();
-            if r < 0 {
-                Err(r.into())
-            } else {
-                Ok(Fs(()))
-            }
```

Comment:
Do we still need `fsInit` for `std::fs` stuff to work, or does `libctru` initialize it for us? I guess `romfs:/` already has its own service, but do `sdmc:/` files still work after this change?

Reply:
Yes, that work is done automatically by `libctru`.